sklearn/tests/test_discriminant_analysis.py

```diff
@@ -350,3 +350,15 @@ def test_covariance():
 
     c_s = _cov(x, 'auto')
     assert_almost_equal(c_s, c_s.T)
+
+
+def test_raises_value_error_on_same_number_of_classes_and_samples():
+    """
+    Tests that if the number of samples equals the number
+    of classes, a ValueError is raised.
+    """
+    no_classes = 2
+    X = np.zeros((no_classes, no_classes))
+    y = np.zeros((no_classes, 1))
+    clf = LinearDiscriminantAnalysis(solver="svd")
+    assert_raises(ValueError, clf.fit, X, y)
```

Comment:
```suggestion
    with pytest.raises(ValueError, match="The number of samples must be more"):
        clf.fit(X, y)
```

Reply:
ah nice, tests the message too to make sure that its actually THAT ValueError being raised

Reply:
yes, and since the adoption of `pytest`, `pytest....` are generally preferred to custom `assert_...` ones.

Reply:
thanks for your advice!